src/System.Memory/src/System/Buffers/ReadOnlySequence_helpers.cs

```diff
@@ -251,22 +230,18 @@ private long GetLength(in SequencePosition start, in SequencePosition end)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private void BoundsCheck(in SequencePosition start, in SequencePosition end)
         {
-            GetTypeAndIndices(start.GetInteger(), end.GetInteger(), out SequenceType type, out int startIndex, out int endIndex);
-
+            int startIndex = GetIndex(start);
+            int endIndex = GetIndex(end);
             object startObject = start.GetObject();
             object endObject = end.GetObject();
+            SequenceType type = GetSequenceType();
             if (type == SequenceType.MultiSegment && startObject != endObject)
             {
-                Debug.Assert(startObject != null);
```

Comment:
We should still keep the `!= null` Debug.Asserts.

Reply:
I think next casting makes such test too `var startSegment = (ReadOnlySequenceSegment<T>)startObject;`

Reply:
I think you can still cast a null object to `ReadOnlySequenceSegment<T>` without any exception.
```C#
object obj = null;
var segment = (ReadOnlySequenceSegment<byte>)obj; // this will not throw
var mem = startSegment.Memory; // this will throw
```